apps/web/src/app/[locale]/settings/_components/settings.action.ts

```diff
@@ -3,26 +3,36 @@
 import { getServerAuthSession } from '@repo/auth/server';
 import { revalidatePath } from 'next/cache';
 import { prisma } from '@repo/db';
-import type { FormSchema } from '.';
+import type { ProfileSchema } from './schema';
+import { profileSchema } from './schema';
 
 /**
  * This will only let you update your own profile
  * @param profileData
  */
-// TODO: add transactions to this update #GFI
-export async function updateProfile(profileData: FormSchema) {
+export async function updateProfile(profileData: ProfileSchema) {
   const session = await getServerAuthSession();
 
   // 1. Checks that the user is logged in
-  if (!session?.user.id) return 'unauthorized';
+  if (!session?.user.id) return { error: 'unauthorized' };
 
-  // 2. Update the user bio field in the db
+  // 2. test schema validation with zod
+  try {
```

Comment:
can we just throw so we can try catch in consumer. not a fan of having to check error cause it just ends like a fetch we have at home. we also dont need to display the real error to users in most cases so returning isnt needed. in cases that it is we can handle that ad hoc

thoughts @QuiiBz